### PR TITLE
feat(Hoppscotch): add presence

### PR DESCRIPTION
**Pull request by itsmeeudrino**
## Description 

This PR adds the [Hoppscotch](https://hoppscotch.io/) presence service, including:

1. REST APIs (Homepage)
2. GraphQL
3. Realtime: Websocket, SSE, SocketIO, MQTT
4. Settings

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image

</details>

### CODE REVIEW

**Comment by itsmeeudrino** [comment]
I might've went a little overboard with the code structure in this one, please feel free to suggest any changes!

File: websites/H/Hoppscotch/lib/realtime.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export function getRealtimeEndpoint(): string | null {
+	const input = document.querySelector("input[type=url]") as HTMLInputElement;
```
Comment (theusaf) [inline websites/H/Hoppscotch/lib/realtime.ts:2]:
```suggestion
	const input = document.querySelector<HTMLInputElement>("input[type=url]")
```

File: websites/H/Hoppscotch/lib/workspace.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+type Workspace =
+	| { kind: "Personal" }
+	| { kind: "Team"; name: string }
+	| { kind: "Error" };
+
+const getWorkspace = (): Workspace => {
+	// eslint-disable-next-line no-one-time-vars/no-one-time-vars
```
Comment (theusaf) [inline websites/H/Hoppscotch/lib/workspace.ts:7]: Avoid disabling eslint rules unless it breaks how the code works. These rules are there to ensure code consistency across the project.

File: websites/H/Hoppscotch/metadata.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+		"name": "eudrino"
+	},
+	"service": "Hoppscotch",
+	"description": {
+		"en": "Hoppscotch is an open source API development ecosystem"
+	},
+	"url": "hoppscotch.io",
+	"version": "1.0.0",
+	"logo": "https://hoppscotch.io/icon.png",
+	"thumbnail": "https://hoppscotch.io/icon.png",
```
Comment (theusaf) [inline websites/H/Hoppscotch/metadata.json:14]: > The [thumbnail](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#thumbnail) should preferably be a [wide promotional card](https://i.imgur.com/3QfIc5v.jpg) or a [screenshot](https://i.imgur.com/OAcBmwW.png) if the first is not available.

File: websites/H/Hoppscotch/presence.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+export const enum Assets {
+	Logo = "https://hoppscotch.io/icon.png",
+	Rest = "https://i.imgur.com/CKj7VUF.png",
+	GraphQL = "https://i.imgur.com/00szkIv.png",
+	Websocket = "https://i.imgur.com/daZZiEo.png",
+	ServerSentEvents = "https://i.imgur.com/CEL3dNu.png",
+	SocketIO = "https://i.imgur.com/UvlTI6Y.png",
+	MQTT = "https://i.imgur.com/r5NFbHF.png",
+	Settings = "https://i.imgur.com/UxuTIv7.png",
```
Comment (theusaf) [inline websites/H/Hoppscotch/presence.ts:21]: [Assets](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets) should be 512x512px

File: websites/H/Hoppscotch/routes/graphql.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Assets } from "../presence";
+import { getWorkspaceName } from "../lib/workspace";
+
+function getEndpoint() {
+	const input = document.querySelector("input#url") as HTMLInputElement;
```
Comment (theusaf) [inline websites/H/Hoppscotch/routes/graphql.ts:5]:
```suggestion
	const input = document.querySelector<HTMLInputElement>("input#url");
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by itsmeeudrino — COMMENTED** [review]
Thanks for taking the time to look at my PR!

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**